src/node/connection_types.h

```diff
@@ -79,4 +79,14 @@ enum class ConnectionType {
 /** Convert ConnectionType enum to a string value */
 std::string ConnectionTypeAsString(ConnectionType conn_type);
 
+/** Transport layer version */
+enum class TransportProtocolType {
+    DETECTING, //!< Peer could be v1 or v2
+    V1, //!< Unencrypted, plaintext protocol
+    V2, //!< BIP324 protocol
+};
```

Comment:
I don't know why I don't see this warning when compiling, but only inside youcompleteme. Both should be using the same flags. Anyway, mentioning it here:

> src/node/connection_types.h|17 col 12 error| Enum 'ConnectionType' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-  enum-size]